src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessCollectionTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Diagnostics.ProcessTests
+{
+    public class ProcessCollectionTests : ProcessTestBase
+    {
+        [Fact]
+        public void TestModuleCollectionBehavior()
+        {
+            ProcessModule[] mArray = _process.Modules.Cast<ProcessModule>().ToArray();
+
+            ProcessModuleCollection moduleCollection = new ProcessModuleCollection(mArray);
+            Assert.True(moduleCollection.Count > 1);
```

Comment:
(Also, doesn't this suffer from the same problem we've hit elsewhere, that the mArray could be empty if we're at a bad point during the target Process' startup?)